src/app/core/services/label.service.ts

```diff
@@ -0,0 +1,141 @@
+import { Injectable } from '@angular/core';
+import {GithubService} from './github.service';
+import {map} from 'rxjs/operators';
+import {Label} from '../models/label.model';
+import { Observable } from 'rxjs';
+import {SEVERITY_ORDER} from '../../core/models/issue.model';
+import { ErrorHandlingService } from './error-handling.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LabelService {
+  private severityLabels: Label[];
+  private typeLabels: Label[];
+  private responseLabels: Label[];
+  private duplicateLabelMap: Map<string, boolean>;
+
+
+  constructor(private githubService: GithubService, private errorHandlingService: ErrorHandlingService) {
+    this.severityLabels = new Array();
+    this.typeLabels = new Array();
+    this.responseLabels = new Array();
+    this.duplicateLabelMap = new Map();
+  }
+
+  /**
+   * Calls the Github service api to get all labels from the repository and
+   * store it in a list of arrays in this label service
+   */
+  getAllLabels(): Observable<void> {
+    try {
```

Comment:
I dont think you can catch the error like this. You need to handle the error at the place you call subscribe().

Reply:
Got it, I have move it to the appropriate location

Reply:
This is rxjs's way of "catching" the error whenever you subscribe to an Observable. 

https://www.learnrxjs.io/operators/error_handling/catch.html